code/ecdsap256sha2/Hacl.Impl.ECDSA.P256SHA256.Verification.Agile.fst

```diff
@@ -88,29 +95,50 @@ let ecdsa_verification_step1 r s =
   let isSCorrect = isMoreThanZeroLessThanOrderMinusOne s in
   isRCorrect && isSCorrect
 
-
-inline_for_extraction noextract
-val ecdsa_verification_step23: hashAsFelem : felem -> mLen: size_t -> m: lbuffer uint8 mLen -> Stack unit
-  (requires fun h -> live h hashAsFelem /\ live h m)
-  (ensures  fun h0 _ h1 -> modifies (loc hashAsFelem) h0 h1 /\
-  (
+inline_for_extraction
+val ecdsa_verification_step23: alg: hash_alg {SHA2_256? alg \/ SHA2_384? alg \/ SHA2_512? alg (* and blake one day *)}
```

Comment:
since this refinement was already used in the type of ecdsa_signature_step12, I would suggest:
```
let is_supported_alg = function | SHA2_256 | SHA2_284 | SHA2_512 -> true | _ -> false
let supported_hash_alg = a:hash_alg { is_supported_alg a }
```